src/devices/Tca954x/Tca9548A.cs

```diff
@@ -0,0 +1,196 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Device.I2c;
+using System.Device.Model;
+using System.IO;
+
+namespace Iot.Device.Tca954x
+{
+    /// <summary>
+    /// Tca9548A - 8-Channel I2C Switch/Multiplexer
+    /// </summary>
+    [Interface("Tca9548A - 8-Channel I2C Switch/Multiplexer")]
+    public class Tca9548A : IList<I2cBus>, IDisposable
+    {
+        private readonly bool _shouldDispose;
+        private readonly List<Tca9548AChannelBus> _channelBuses = new List<Tca9548AChannelBus>();
+        private MultiplexerChannel? _activeChannels;
+
+        /// <summary>
+        /// The default I2C Address, page 15 of the main documentation
+        /// https://www.ti.com/lit/ds/symlink/tca9548a.pdf
+        /// </summary>
+        public const byte DefaultI2cAddress = 0x70;
+
+        /// <summary>
+        /// Array of all possible Multiplexer Channels
+        /// </summary>
+        private static readonly MultiplexerChannel[] DeviceChannels = new MultiplexerChannel[]
+        {
+            MultiplexerChannel.Channel0, MultiplexerChannel.Channel1, MultiplexerChannel.Channel2, MultiplexerChannel.Channel3, MultiplexerChannel.Channel4,
+            MultiplexerChannel.Channel5, MultiplexerChannel.Channel6, MultiplexerChannel.Channel7
+        };
+
+        internal I2cDevice _i2CDevice;
```

Comment:
for internal, isn't it better to use a property?

Reply:
Changed this to private. Can think of making it a property when we introduce derived classes or siblings.

Reply:
Oh, good you found this one - looking at this, just found a major bug in the channel implementation.